src/common/dns_utils.cpp

```diff
@@ -444,6 +445,78 @@ DNSResolver DNSResolver::create()
 namespace dns_utils
 {
 
+uint32_t swap_byte_order( uint32_t hostlong)
+{
+  union {
+      uint32_t integer;
+      uint8_t bytes[4];
+  } value { hostlong };
+
+  std::swap(value.bytes[0], value.bytes[3]);
+  std::swap(value.bytes[1], value.bytes[2]);
+
+  return value.integer;
+}
+
+uint32_t ipaddr_ipv4_bits(const uint8_t a, const uint8_t b, const uint8_t c, const uint8_t d)
+{
+  return ( (a << 24) | (b << 16) | (c << 8) | d );
+}
+
+uint32_t netmask_to_cidr(uint32_t netmask)
+{
+  int cidr = 0;
+  while ( netmask )
+  {
+      cidr += ( netmask & 0x01 );
+      netmask >>= 1;
+  }
+  return cidr;
+}
+
+uint32_t netmask_ipv4_bits(int prefix)
+{
+	if (prefix) {
+		return ~((1 << (32 - prefix)) - 1);
+	} else {
+		return uint32_t{0};
+	}
+}
+
+std::tuple<uint32_t, uint32_t> FromIPv4(const uint8_t a, const uint8_t b, const uint8_t c, const uint8_t d, const uint32_t netmask)
+{
+  return std::tuple<uint32_t,uint32_t>{ipaddr_ipv4_bits(a,b,c,d), netmask_ipv4_bits(netmask)};
+}
+
+std::array bogonRanges = {FromIPv4(0, 0, 0, 0, 8),
+                           FromIPv4(10, 0, 0, 0, 8),
+                           FromIPv4(21, 0, 0, 0, 8),
```

Comment:
`21.0.0.0/8` was a DoD internal range but it is soon being sold and will be publically rotatable in the future.